torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -774,6 +925,10 @@ def _setup_streams(self) -> None:
             self._streams["all_gather"] = torch.cuda.Stream()
             # Stream for overlapping grad reduction with the backward pass.
             self._streams["post_backward"] = torch.cuda.Stream()
+            # Stream to move main params to self.mixed_precision.param_dtype
+            # for forward pass.
+            if self.mixed_precision:
```

Comment:
Yes, the main purpose is when the main param is on CPU and we have a .to() to get the param to GPU before copying it into the reduced precision tensor. 

It is similar to what we do for DDP: 
https://github.com/pytorch/pytorch/blob/master/torch/nn/parallel/distributed.py#L1069

Performance-wise, we can test the following as follow up - 
1) If we really need this stream, or if there is no increase in overhead when running MP on large models such as GPT
2) Whether it would be useful to consider performing CPU -> GPU copies in a separate stream for non mixed precision case (we currently do this within the all_gather stream).